core/src/main/java/org/elasticsearch/search/query/QueryPhase.java

```diff
@@ -217,7 +217,9 @@ public TopDocs call() throws Exception {
                         }
                     }
                 } else {
-                    lastEmittedDoc = null;
+                    if (searchContext.searchAfter() != null) {
+                        after = searchContext.searchAfter();
+                    }
```

Comment:
do we need the outer `if` statement or could we just do `after = searchContext.searchAfter();`?